### PR TITLE
Show active and total member joint to the group

### LINKED ISSUE #27: Add members active and total in sign in login
Use web socket api

- [ ] Show the members active from general channel

- [ ] Show the total of member from general channel

### PR DESCRIPTION
**Pull request by Linzeur**
- Close #27 

- In login file:
    - Create a global variable named `usersJoint` assigned with an empty array, which will store the 
      information of all connected user at that moment.
    - Create a function `reconnectServer` to reconnect when the socket is closed.
    - Create a function `askConnectedMembers` which is executed when the open event from socket is 
      triggered. It sends to all connected users a specific message. The message is sent each specific 
      time which is a constant variable and now it is defined with 3 seconds.
    - Create a function `getNumberConnectedMembers` which is executed when the message event from 
      socket is triggered. It receives a message which is an answer to the message sent by previous 
      function. The message is a json object with an unique key and its value is an array. It value is 
      assigned to the global variable previously commented, and it show in the page the values 
      respectively
    - Modify the function `newAppObject` it creates the general object and add the information from 
      global variable `usersJoint` to the key `users`.

- In index file:
     - Create a new function `reconnectServer` to refactor the anonymous function of the close event from 
       socket.
     - Add event `beforeunload` to the window to send a message which indicate if an user already 
       connected is disconnecting.
     - Create  function `modifyStateUsers` which modify the key `isActive` of one specific user
     - Create function `filterOwnMessages` which filter all message received in message event from 
       socket with property `isNotification` equal to true.
     - Create function `receiveMessages` which receive all messages emitted among users within a 
       channel.
     - Modify the anonymous function of the message event from socket to difference the messages 
       received since some of them are to verify how many users are connected and other to emit 
       messages.

### CODE REVIEW

File: public/assets/js/index.js [hunk 1]
```diff
@@ -172,50 +175,95 @@ function listAllMessages() {
...
+
+function modifyStateUsers(idUser, newState) {
+  app.users = app.users.map(currentValue => {
+    if (currentValue.id == idUser) currentValue.isActive = newState;
+    return currentValue;
+  });
+}
+
+function receiveMessages(data) {
+  if (validateOwnMessages(data)) {
```
Comment (diegotc86) [inline public/assets/js/index.js]: Hi, this function should be `filterOwnMessages`, rigth?

File: public/assets/js/index.js [hunk 2]
```diff
@@ -172,50 +175,95 @@ function listAllMessages() {
...
+
+      if (data.content.indexOf("has joint to this group") > -1) {
+        let newUserJoint = data.author;
+        newUserJoint["isActive"] = true;
+        app.users = app.users.concat(newUserJoint);
+      } else modifyStateUsers(data.author.id, true);
+    }
+
+    console.log(data);
+    app.channels[indexChannelActive].messages.push(datay);
```
Comment (diegotc86) [inline public/assets/js/index.js]: Typo here!

File: public/assets/js/index.js [hunk 3]
```diff
@@ -172,50 +175,95 @@ function listAllMessages() {
...
     let newMessage = createNewMessage(messageForAll, user, true);
+
     socket.send(JSON.stringify(newMessage));
     localStorage.setItem(keyStorage, JSON.stringify(app));
     if (firstConnection) listAllMessages();
     firstConnection = false;
   }
 }
 
+function reconnectServer() {
```
Comment (paulotijero) [inline public/assets/js/index.js]: This is an elegant function, very nice! :+1:

File: public/assets/js/index.js (hunk 1 above)
Comment (Linzeur) [reply public/assets/js/index.js]: Oh right, I fixed that! Thanks

File: public/assets/js/index.js (hunk 2 above)
Comment (Linzeur) [reply public/assets/js/index.js]: Thanks!! I didn't realize that

**Pull request merged**